sbi/neural_nets/net_builders/categorial.py

```diff
@@ -63,9 +63,12 @@ def build_categoricalmassestimator(
             unique(col).numel() for col in batch_x_discrete.T
         ])
         num_categories = inferred_categories
+    # get all the unique values for each column
+    categorical_values = [unique(col) for col in batch_x.T]
 
     categorical_net = CategoricalMADE(
         num_categories=num_categories,
+        categorical_values=categorical_values,
```

Comment:
I see, here we do this automatically, so the user don't have to change anything. but one could argue that it's just the responsibility of the `CategoricalMADE` to deal with the value-index mapping?